src/test/java/org/broadinstitute/hellbender/tools/coveragemodel/cachemanager/ComputableGraphStructureUnitTest.java

```diff
@@ -1,19 +1,359 @@
+                .computableNode("g", new String[] {}, new String[] {"y", "z"}, null, true)
+                .computableNode("h", new String[] {}, new String[] {"f", "g"}, null, true)
+                .build();
+    }
+
+    @Test(expectedExceptions = ComputableGraphStructure.CyclicGraphException.class)
+    public void testCyclicGraphException_1() {
+        ImmutableComputableGraph.builder()
+                .primitiveNode("x", new String[] {}, new DuplicableNDArray())
+                .computableNode("y", new String[] {}, new String[] {"x", "w"}, null, true) /* cycle */
+                .computableNode("z", new String[] {}, new String[] {"y"}, null, true)
+                .computableNode("w", new String[] {}, new String[] {"z"}, null, true)
+                .build();
+    }
+
+    @Test(expectedExceptions = ComputableGraphStructure.CyclicGraphException.class)
+    public void testCyclicGraphException_2() {
+        ImmutableComputableGraph.builder()
+                .primitiveNode("x", new String[] {}, new DuplicableNDArray())
+                .primitiveNode("y", new String[] {}, new DuplicableNDArray())
+                .primitiveNode("z", new String[] {}, new DuplicableNDArray())
+                .computableNode("f", new String[] {}, new String[] {"x", "y", "h"}, null, true) /* cycle */
+                .computableNode("g", new String[] {}, new String[] {"y", "z"}, null, true)
+                .computableNode("h", new String[] {}, new String[] {"f", "g"}, null, true)
+                .build();
+    }
+
+    @Test(invocationCount = NUM_TRIALS)
+    public void testNodeTagsAndKeys() {
+        final RandomDAG dag = RandomDAG.getRandomDAG();
+        final ComputableGraphStructure cgs = new ComputableGraphStructure(dag.getEquivalentCacheNodeSet());
+        final Set<String> cgsNodeTagsSet = cgs.getNodeTagsSet();
+        final Set<String> dagNodeTagsSet = dag.tagsSet;
+        final Set<String> cgsNodeKeysSet = cgs.getNodeKeysSet();
+        final Set<String> dagNodeKeysSet = dag.nodeKeysSet;
+        Assert.assertTrue(cgsNodeKeysSet.equals(dagNodeKeysSet));
+        Assert.assertTrue(cgsNodeTagsSet.equals(dagNodeTagsSet));
+    }
+
+    @Test(invocationCount = NUM_TRIALS)
+    public void testTopologicalOrder() {
+        final RandomDAG dag = RandomDAG.getRandomDAG();
+        final ComputableGraphStructure cgs = new ComputableGraphStructure(dag.getEquivalentCacheNodeSet());
+        dag.nodeKeysSet.forEach(nodeKey -> Assert.assertTrue(dag.topologicalOrderMap.get(nodeKey) ==
+                cgs.getTopologicalOrder(nodeKey)));
+    }
+
+    @Test(invocationCount = NUM_TRIALS)
+    public void testAncestors() {
+        final RandomDAG dag = RandomDAG.getRandomDAG();
+        final ComputableGraphStructure cgs = new ComputableGraphStructure(dag.getEquivalentCacheNodeSet());
+        dag.nodeKeysSet.forEach(nodeKey -> Assert.assertTrue(dag.getAncestors(nodeKey).equals(cgs.getAncestors(nodeKey))));
```

Comment:
I think smaller tests, if possible, are always preferred (for faster localization of failing tests).